### PR TITLE
feat(virtual-scroll): usage with custom scroll target

### PR DESCRIPTION
**Pull request by sean-perkins**
Updates the sections for note regarding Ionic components to include the usage instructions for the custom scroll target.

### CODE REVIEW

File: docs/react/virtual-scroll.md [hunk 1]
```diff
@@ -57,6 +57,20 @@ Certain Ionic Framework functionality is currently not compatible with virtual s
...
+## Usage with Ionic Components
+
+Ionic Framework requires that features such as collapsible large titles, `ion-infinite-scroll, `ion-refresher`and`ion-reorder-group`be used within an`ion-content`. To use these experiences with virtual scrolling, you must add the `.ion-content-scroll-host` class to the virtual scroll viewport.
+
+For example:
+
+```tsx
+<IonPage>
+  <IonContent scrollY={false}>
+    <Virtuoso className="ion-content-scroll-host">// Your existing content and configurations</Virtuoso>
```
Comment (amandaesmith3) [inline docs/react/virtual-scroll.md:69]:
```suggestion
    <Virtuoso className="ion-content-scroll-host">
      {/* Your existing content and configurations */}
    </Virtuoso>
```

**Review by amandaesmith3 — CHANGES_REQUESTED** [review]

File: docs/angular/virtual-scroll.md [hunk 2]
```diff
@@ -110,11 +110,19 @@ cdk-virtual-scroll-viewport {
...
+## Usage with Ionic Components
 
-Certain Ionic Framework functionality is currently not compatible with virtual scrolling. Features such as collapsible large titles, `ion-infinite-scroll`, and `ion-refresher` rely on being able to scroll on `ion-content` itself, and as a result will not work when using virtual scrolling.
+Ionic Framework requires that features such as collapsible large titles, `ion-infinite-scroll, `ion-refresher`and`ion-reorder-group`be used within an`ion-content`. To use these experiences with virtual scrolling, you must add the `.ion-content-scroll-host` class to the virtual scroll viewport.
 
-We are working to improve compatibility between these components and virtual scrolling solutions. You can follow progress and give feedback here: https://github.com/ionic-team/ionic-framework/issues/23437.
+For example:
+
+```html
+<ion-content scroll-y="false">
```
Comment (liamdebeasi) [inline docs/angular/virtual-scroll.md:120]: Now that we are in an Angular specific documentation, we should use the angular syntax here. So `scroll-y` should be `[scrollY]`.

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

**Review by amandaesmith3 — APPROVED** [review]

File: docs/react/virtual-scroll.md [hunk 3]
```diff
@@ -57,6 +57,22 @@ Certain Ionic Framework functionality is currently not compatible with virtual s
...
 We are working to improve compatibility between these components and virtual scrolling solutions. You can follow progress and give feedback here: https://github.com/ionic-team/ionic-framework/issues/23437.
 
+## Usage with Ionic Components
+
+Ionic Framework requires that features such as collapsible large titles, `ion-infinite-scroll`, `ion-refresher`, and `ion-reorder-group` be used within an `ion-content`. To use these experiences with virtual scrolling, you must add the `.ion-content-scroll-host` class to the virtual scroll viewport.
+
+For example:
+
+```tsx
+<IonPage>
```
Comment (liamdebeasi) [inline docs/react/virtual-scroll.md:67]: React has `<IonPage>`, but Vue does not. Can we either remove it from React or add it to Vue?

**Review by liamdebeasi — APPROVED** [review]
Good to merge once my comment below is addressed.

**Comment by lincolnthree** [comment]
Hi all. Please see the following issue in the Angular CDK, as I believe this will make much of this easier if we can get these features merged: https://github.com/angular/components/pull/24394

**Comment by sean-perkins** [comment]
@lincolnthree that feature change would allow developers to make use of the `ion-content`'s scroll container instead of the virtual scroll container, but the implementation code may be more difficult for implementers to follow:

```html
<ion-content #content>
  <cdk-virtual-scroll-viewport [scrollableElement]="content.getScrollElement() | async">
    ...
 </cdk-virtual-scroll-viewport>
</ion-content>
```

I don't see any downsides to that feature change, but we would continue to offer this class approach to support all virtual scrolling solutions in all supported frameworks, not just CDK & Angular.

For example, this is how it looks for React, Vue and Angular with the same implementation pattern: https://gist.github.com/sean-perkins/277524ef0e450591907a87acec15af11

Appreciate your feedback and making us aware of that feature/change from Angular!

**Comment by Albert-Gao** [comment]
with this being merged, i guess this doc is outdated? https://ionicframework.com/docs/react/virtual-scroll#a-note-on-ionic-components :)

**Comment by avalanche1** [comment]
@sean-perkins pls deprecate  https://ionicframework.com/docs/react/virtual-scroll#a-note-on-ionic-components as obsolete

**Comment by liamdebeasi** [comment]
Thanks everyone, I fixed this in https://github.com/ionic-team/ionic-docs/pull/2780.

**Pull request merged**